### PR TITLE
Angular deprecation: Dashboards: Add link to auto-migrate compatible panels to dashboard warning

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

Adds a link to Angular dashboard warnings that points to `&__feature.autoMigrateOldPanels=true`.

It is be displayed only when the dashboard contains panels that can be migrated.

immagine

Requires #84034 to be merged before this PR, in order to set the feature flag from URL in production mode.

**Why do we need this feature?**

Simplify auto-migrating Angular panels.

**Who is this feature for?**

Grafana users with Angular panel plugins.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Need to add more tests

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by lukasztyrala** [comment]
Hi @xnyo, at DataViz team we are trying to put together a script (guidelines) to use when communicating with customers about the expected deprecations and how to safely test the auto migration. There are few things we need to understand:
1. Do we know how we will integrate the migration button into the UI and where the banner will be displayed?
2. Will there be an option to revert the automigration (or will it still be required to save dashboard explicitly after the auto migration)?

cc @nmarrs @leeoniya

**Comment by xnyo** [comment]
Hi @lukasztyrala 

> Do we know how we will integrate the migration button into the UI and where the banner will be displayed?

The link displayed in the screenshot is shown in a warning that's displayed at the top of dashboards that use Angular plugins. This warning is displayed since Grafana 10.4.0. The "Auto-migrate compatible panels" link in particular will be displayed when the dashboard contains at least one of the following plugins:

- graph
- table-old
- grafana-piechart-panel
- grafana-worldmap-panel
- singlestat
- grafana-singlestat-panel

Here's a better screenshot showing it:

image


> Will there be an option to revert the automigration (or will it still be required to save dashboard explicitly after the auto migration)?

I don't know the details regarding the auto migration feature, but from my testing the dashboard required a manual save before the migration was committed. So there's no option to revert the auto migration because it has to be saved manually by the user

**Comment by leeoniya** [comment]
> So there's no option to revert the auto migration because it has to be saved manually by the user

yes, once it is saved there is no going back, unless you restore old database or if we offer a built-in way to undo db changes (such as retaining old versions somewhere else in the db). but this will only work until angular is removed anyhow.

i think the main issue with the inability to use a url for feature enablement in production environments is that to "undo" an unsaved migration is no longer a browser Back button away. we need to provide a way to clear the appropriate cookie or localStorage variable that we choose to set instead of relying on browser url history.

**Comment by sympatheticmoose** [comment]
>yes, once it is saved there is no going back, unless you restore old database or if we offer a built-in way to undo db changes (such as retaining old versions somewhere else in the db). but this will only work until angular is removed anyhow.

Couldn't you just restore the previous dashboard version? Or is the change persisted somewhere else?
Screenshot 2024-02-29 at 17 33 45

**Comment by leeoniya** [comment]
> Couldn't you just restore the previous dashboard version? Or is the change persisted somewhere else beyond the dashboard.json?

the dashboard.json is stored in the db. so if we provide some way to restore previous dashboard versions, then sure. this is outside my wheelhouse. idk what we offer in terms of backend storage capabilities or how/where we expose this in the UI.

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 1]
```diff
@@ -121,6 +121,15 @@ export class DashboardModel implements TimeModel {
...
   };
 
+  static autoMigratePluginIDs = [
+    'graph',
+    'table-old',
+    'grafana-piechart-panel',
+    'grafana-worldmap-panel',
+    'singlestat',
+    'grafana-singlestat-panel',
+  ];
```
Comment (nmarrs) [inline public/app/features/dashboard/state/DashboardModel.ts:131]: This list already exists in the code here :) https://github.com/grafana/grafana/blob/01fb2cff624e402b927e16f4e04e9fc35c7ab08c/public/app/features/dashboard/state/PanelModel.ts#L133

Comment (xnyo) [reply public/app/features/dashboard/state/DashboardModel.ts]: Spot on :+1: Fixed in 3ce974089bffbed30c9f4c65fece85547152b2be

**Comment by xnyo** [comment]
Superseded by #84100

**Pull request closed without merging**